sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -61,21 +70,30 @@ internal partial class BlobsCheckpointStore : StorageManager
         ///
         private EventHubsRetryPolicy RetryPolicy { get; }
 
+        /// <summary>
+        ///   Indicates whether to read legacy checkpoints when no current version checkpoints are available.
+        /// </summary>
+        ///
+        private bool ReadLegacyCheckpoints { get; }
+
         /// <summary>
         ///   Initializes a new instance of the <see cref="BlobsCheckpointStore" /> class.
         /// </summary>
         ///
         /// <param name="blobContainerClient">The client used to interact with the Azure Blob Storage service.</param>
         /// <param name="retryPolicy">The retry policy to use as the basis for interacting with the Storage Blobs service.</param>
+        /// <param name="readLegacyCheckpoints">Indicates whether to read legacy checkpoints when no current version checkpoints are available.</param>
```

Comment:
I'm not clear on the semantics from the comment.  Does that apply to an individual partition or all-up?    For example, if I have a current checkpoint for partition "0" but none for partition "1" and I have legacy checkpoints for both... do I see the legacy checkpoint for "1" or no checkpoint?